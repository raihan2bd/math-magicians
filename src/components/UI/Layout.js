import React from 'react';
import PropTypes from 'prop-types';

import Header from './Header';
import classes from './Layout.module.css';

const Layout = (props) => {
  const { children } = props;
  return (
    <>
      <Header />
      <main className={classes.main}>
        <div className={classes.container}>{children}</div>
      </main>
    </>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
