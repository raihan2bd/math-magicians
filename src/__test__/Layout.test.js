import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import Layout from '../components/UI/Layout';

describe('Test Layout with snapshot', () => {
  test('shoud match with snapshot', () => {
    const layout = renderer.create(
      <BrowserRouter>
        <Layout>
          <h1>Hello world</h1>
        </Layout>
      </BrowserRouter>,
    );

    expect(layout).toMatchSnapshot();
  });

  test('shoud render child component', () => {
    render(
      <BrowserRouter>
        <Layout>
          <h1>Hello world</h1>
        </Layout>
      </BrowserRouter>,
    );

    expect(screen.getByText('Hello world')).toBeInTheDocument();
  });
});
