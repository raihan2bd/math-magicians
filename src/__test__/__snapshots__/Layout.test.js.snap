// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Layout with snapshot shoud match with snapshot 1`] = `
Array [
  <header
    className="header"
  >
    <nav
      className="nav"
    >
      <div
        className="nav_brand"
      >
        <h1>
          <a
            href="/"
            onClick={[Function]}
          >
            Math Magicians
          </a>
        </h1>
      </div>
      <ul
        className="nav_group"
      >
        <li
          className="nav_item"
        >
          <a
            aria-current="page"
            className="nav_link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="nav_link"
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            className="nav_link"
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    className="main"
  >
    <div
      className="container"
    >
      <h1>
        Hello world
      </h1>
    </div>
  </main>,
]
`;

exports[`Test Layout with snapshot shoud render child component 1`] = `
<h1>
  Hello world
</h1>
`;
